perception/object_recognition/detection/roi_cluster_fusion/src/roi_cluster_fusion_nodelet.cpp

```diff
@@ -243,30 +243,34 @@ void RoiClusterFusionNodelet::cameraInfoCallback(
 }
 
 void RoiClusterFusionNodelet::fusionCallback(
-  autoware_perception_msgs::msg::DynamicObjectWithFeatureArray::ConstSharedPtr input_cluster_msg,
-  autoware_perception_msgs::msg::DynamicObjectWithFeatureArray::ConstSharedPtr input_roi0_msg,
-  autoware_perception_msgs::msg::DynamicObjectWithFeatureArray::ConstSharedPtr input_roi1_msg,
-  autoware_perception_msgs::msg::DynamicObjectWithFeatureArray::ConstSharedPtr input_roi2_msg,
-  autoware_perception_msgs::msg::DynamicObjectWithFeatureArray::ConstSharedPtr input_roi3_msg,
-  autoware_perception_msgs::msg::DynamicObjectWithFeatureArray::ConstSharedPtr input_roi4_msg,
-  autoware_perception_msgs::msg::DynamicObjectWithFeatureArray::ConstSharedPtr input_roi5_msg,
-  autoware_perception_msgs::msg::DynamicObjectWithFeatureArray::ConstSharedPtr input_roi6_msg,
-  autoware_perception_msgs::msg::DynamicObjectWithFeatureArray::ConstSharedPtr input_roi7_msg)
+  autoware_perception_msgs::msg::DetectedObjectsWithFeature::ConstSharedPtr input_cluster_msg,
+  autoware_perception_msgs::msg::DetectedObjectsWithFeature::ConstSharedPtr input_roi0_msg,
+  autoware_perception_msgs::msg::DetectedObjectsWithFeature::ConstSharedPtr input_roi1_msg,
+  autoware_perception_msgs::msg::DetectedObjectsWithFeature::ConstSharedPtr input_roi2_msg,
+  autoware_perception_msgs::msg::DetectedObjectsWithFeature::ConstSharedPtr input_roi3_msg,
+  autoware_perception_msgs::msg::DetectedObjectsWithFeature::ConstSharedPtr input_roi4_msg,
+  autoware_perception_msgs::msg::DetectedObjectsWithFeature::ConstSharedPtr input_roi5_msg,
+  autoware_perception_msgs::msg::DetectedObjectsWithFeature::ConstSharedPtr input_roi6_msg,
+  autoware_perception_msgs::msg::DetectedObjectsWithFeature::ConstSharedPtr input_roi7_msg)
 {
   // Guard
   if (labeled_cluster_pub_->get_subscription_count() < 1) {
     return;
   }
 
   // build output msg
-  autoware_perception_msgs::msg::DynamicObjectWithFeatureArray output_msg;
-  output_msg = *input_cluster_msg;
+  autoware_auto_perception_msgs::msg::DetectedObjects output_msg;
+  output_msg.header = input_cluster_msg->header;
+  for (const auto & feature_object : input_cluster_msg->feature_objects) {
+    output_msg.objects.push_back(feature_object.object);
+  }
```

Comment:
@yukkysaito @yukke42  please check the following modifications :pray: 

- change input topic type DynamicObjectWithFeatureArray -> DetectedObjectsWithFeature
- change output topic type DynamicObjectWithFeatureArray -> DetectedObjects
- remove autoware_perception_msgs/Feature from DetectedObjectsWithFeature in this process

Reply:
fix output topic type in https://github.com/tier4/AutowareArchitectureProposal.iv/pull/536/commits/fb8c160d15f031962b06b6cac48d56098b7083d5

- DetectedObjects -> DynamicObjectWithFeatureArray

Reply:
LGTM